pkg/transform/project_transform.go

```diff
@@ -0,0 +1,157 @@
+type ProjectExtraction struct {
+	legacyconfigv1.MasterConfig
+}
+
+// ProjectTransform is a Project specific transform
+type ProjectTransform struct {
+}
+
+// Transform converts data collected from an OCP3 into a useful output
+func (e ProjectExtraction) Transform() ([]Output, error) {
+	logrus.Info("ProjectTransform::Transform")
+	manifests, err := e.buildManifestOutput()
+	if err != nil {
+		return nil, err
+	}
+	reports, err := e.buildReportOutput()
+	if err != nil {
+		return nil, err
+	}
+	outputs := []Output{manifests, reports}
+	return outputs, nil
+}
+
+func (e ProjectExtraction) buildManifestOutput() (Output, error) {
+	var manifests []Manifest
+
+	projectCR, err := project.Translate(e.MasterConfig.ProjectConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	projectCRYAML, err := GenYAML(projectCR)
+	if err != nil {
+		return nil, err
+	}
+
+	manifest := Manifest{Name: "100_CPMA-cluster-config-project.yaml", CRD: projectCRYAML}
+	manifests = append(manifests, manifest)
+
+	return ManifestOutput{
+		Manifests: manifests,
+	}, nil
+}
+
+func (e ProjectExtraction) buildReportOutput() (Output, error) {
+	reportOutput := ReportOutput{
+		Component: ProjectComponentName,
+	}
+
+	reportOutput.Reports = append(reportOutput.Reports,
+		Report{
+			Name:       "ProjectRequestMessage",
+			Kind:       "ProjectConfig",
+			Supported:  true,
+			Confidence: HighConfidence,
+			Comment:    "Networks must be configured during installation",
+		})
+
+	reportOutput.Reports = append(reportOutput.Reports,
+		Report{
+			Name:       "ProjectRequestTemplate",
+			Kind:       "ProjectConfig",
+			Supported:  true,
+			Confidence: HighConfidence,
+			Comment:    "Networks must be configured during installation",
+		})
+
+	reportOutput.Reports = append(reportOutput.Reports,
+		Report{
+			Name:       "DefaultNodeSelector",
+			Kind:       "ProjectConfig",
+			Supported:  false,
+			Confidence: NoConfidence,
+			Comment:    "Not supported in OCP4: " + e.ProjectConfig.DefaultNodeSelector,
+		})
+
+	reportOutput.Reports = append(reportOutput.Reports,
+		Report{
+			Name:       "SecurityAllocator.mcsAllocatorRange",
+			Kind:       "ProjectConfig",
+			Supported:  false,
+			Confidence: NoConfidence,
+			Comment:    "Not supported in OCP4: " + e.ProjectConfig.SecurityAllocator.MCSAllocatorRange,
+		})
+
+	reportOutput.Reports = append(reportOutput.Reports,
+		Report{
+			Name:       "SecurityAllocator.mcsLabelsPerProject",
+			Kind:       "ProjectConfig",
+			Supported:  false,
+			Confidence: NoConfidence,
+			Comment:    "Not supported in OCP4: " + strconv.Itoa(e.ProjectConfig.SecurityAllocator.MCSLabelsPerProject),
+		})
+
+	reportOutput.Reports = append(reportOutput.Reports,
+		Report{
+			Name:       "SecurityAllocator.uidAllocatorRange",
+			Kind:       "ProjectConfig",
+			Supported:  false,
+			Confidence: NoConfidence,
+			Comment:    "Not supported in OCP4: " + string(e.ProjectConfig.SecurityAllocator.UIDAllocatorRange),
```

Comment:
Golang way of concatenating strings is `sprintf`, but that also looks ok for me after working with `Ruby`

Reply:
I remember reading about such practice, but you're right `Ruby` habit has taken over!
I've changed it. Thank you.